File: src/main/java/ru/vito/desktop/app/models/DefinedUsers.java
package ru.vito.desktop.app.models;

import lombok.Builder;
import lombok.Data;

import java.io.Serializable;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

@Data
@Builder
public class DefinedUsers implements Serializable {

    private Map<String, String> emailToLoginMap;

    public DefinedUsers() {
        this.emailToLoginMap = new HashMap<>();
    }

    public DefinedUsers(final Map<String, String> emailToLoginMap) {
        this.emailToLoginMap = emailToLoginMap;
    }

    @Override
    public String toString() {

        final HashMap<String, Set<String>> loginEmailsMap = new HashMap<>();
        emailToLoginMap.forEach((email, login) -> {

            final Set<String> emails = loginEmailsMap.containsKey(login)
                    ? loginEmailsMap.get(login) : new HashSet<>();
            emails.add(email);

            loginEmailsMap.put(login, emails);
        });


        final StringBuilder builder = new StringBuilder();
        loginEmailsMap.forEach((login, emails) ->
                builder.append(login)
                        .append(" -> ")
                        .append(emails.toString()
                                .replace("[", "")
                                .replace("]", ""))
                        .append("\n"));

        return builder.toString();
    }
}

File: src/main/kotlin/ru/vito/desktop/app/models/DefinedUsers.kt
package ru.vito.desktop.app.models

import java.io.Serializable
import java.util.*

data class DefinedUsers(val emailToLoginMap: Map<String, String>) : Serializable {

    override fun toString(): String {

        val loginEmailsMap = HashMap<String, HashSet<String>?>()

        emailToLoginMap.forEach { email, login ->
            val emails: HashSet<String>? =
                    if (loginEmailsMap.containsKey(login)) loginEmailsMap[login]
                    else HashSet()

            emails?.add(email)

            loginEmailsMap[login] = emails
        }

        val builder = StringBuilder()
        loginEmailsMap.forEach { login, emails ->
            builder.append(login)
                    .append(" -> ")
                    .append(emails.toString()
                            .replace("[", "")
                            .replace("]", ""))
                    .append("\n")
        }

        return builder.toString()
    }
}
